@Preyansh91 you can find the errors in the f5-common-python slack team in the #github-events channel. I'm re-pasting here for convenience.

```
f5/bigip/tm/sys/test/functional/test_version.py::TestVersion::test_entry ERROR

==================================== ERRORS ====================================
_________________ ERROR at teardown of TestVersion.test_entry __________________

    def teardown():
>       v.entries = entries
E       NameError: free variable 'entries' referenced before assignment in enclosing scope

f5/bigip/tm/sys/test/functional/test_version.py:19: NameError
=================================== FAILURES ===================================
____________________________ TestVersion.test_entry ____________________________

self = <f5.bigip.tm.sys.test.functional.test_version.TestVersion object at 0x7f514ada7390>
request = <FixtureRequest for <Function 'test_entry'>>
mgmt_root = <f5.bigip.ManagementRoot object at 0x7f514b33b310>

    def test_entry(self, request, mgmt_root):
        # Load
>       ver1, orig_entries = setup_version_test(request, mgmt_root)

f5/bigip/tm/sys/test/functional/test_version.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
f5/bigip/tm/sys/test/functional/test_version.py:23: in setup_version_test
    entries = v.entries
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

container = <f5.bigip.tm.sys.version.Version object at 0x7f514ada7090>
name = 'entries'

    def __getattr__(container, name):
        # ensure this object supports lazy attrs.
        cls_name = container.__class__.__name__
        if 'allowed_lazy_attributes' not in container._meta_data:
            error_message = ('"allowed_lazy_attributes" not in',
                             'container._meta_data for class %s' % cls_name)
>           raise LazyAttributesRequired(error_message)
E           LazyAttributesRequired: ('"allowed_lazy_attributes" not in', 'container._meta_data for class Version')

f5/bigip/mixins.py:95: LazyAttributesRequired
!!!!!!!!!!!!!!!!!!!! Interrupted: stopping after 2 failures !!!!!!!!!!!!!!!!!!!!
========= 1 failed, 158 passed, 42 skipped, 1 error in 394.09 seconds ==========
ERROR: InvocationError: '/var/lib/jenkins/workspace/f5-sdk/ci.f5.f5-common-python/.tox/py27-bip-v11.5.4/bin/py.test --bigip localhost --port 10443 -s -vv --release 11.5.4 -x --port 11154 f5/bigip/tm/sys/test/functional f5/bigip/tm/sys/test/unit/test_version.py'
___________________________________ summary ____________________________________
ERROR:   py27-bip-v11.5.4: commands failed
Build step 'Execute shell' marked build as failure
Finished: FAILURE
```

--- a/f5/bigip/tm/sys/test/functional/test_version.py
+++ b/f5/bigip/tm/sys/test/functional/test_version.py
@@ -15,14 +15,17 @@
 
 
 def setup_version_test(request, mgmt_root):
+    def teardown():
+        v.selfLink = selflink
+    request.addfinalizer(teardown)
     v = mgmt_root.tm.sys.version.load()
-    entries = v.entries
-    return v, entries
+    selflink = v.selfLink
+    return v, selflink
 
 
 class TestVersion(object):
     def test_entry(self, request, mgmt_root):
         # Load
-        ver1, orig_entries = setup_version_test(request, mgmt_root)
+        ver1, orig_link = setup_version_test(request, mgmt_root)
         ver2 = mgmt_root.tm.sys.version.load()
-        assert len(ver1.entries) == len(ver2.entries)
+        assert len(ver1.selfLink) == len(ver2.selfLink)

--- a/f5/bigip/tm/sys/test/functional/test_version.py
+++ b/f5/bigip/tm/sys/test/functional/test_version.py
@@ -15,12 +15,8 @@
 
 
 def setup_version_test(request, mgmt_root):
-    def teardown():
-        v.entries = entries
-        v.update()
-    request.addfinalizer(teardown)
     v = mgmt_root.tm.sys.version.load()
-    entries = v.entries
+    entries = v.entries    
     return v, entries
 
 

--- a/f5/bigip/tm/sys/test/functional/test_version.py
+++ b/f5/bigip/tm/sys/test/functional/test_version.py
@@ -16,7 +16,7 @@
 
 def setup_version_test(request, mgmt_root):
     v = mgmt_root.tm.sys.version.load()
-    entries = v.entries    
+    entries = v.entries
     return v, entries
 
 

